### PR TITLE
Use proper methods for display name retrieval

### PR DESCRIPTION
**Pull request by MorrisJobke**
* [x] add removal to #26407

### CODE REVIEW

**Review by blizzz — APPROVED** [review]
code looks good

File: apps/files_sharing/lib/Cache.php [hunk 1]
```diff
@@ -166,7 +167,13 @@ protected function formatCacheEntry($entry, $path = null) {
 
 	private function getOwnerDisplayName() {
 		if (!$this->ownerDisplayName) {
-			$this->ownerDisplayName = \OC_User::getDisplayName($this->storage->getOwner(''));
+			$uid = $this->storage->getOwner('');
+			$user = \OC::$server->get(IUserManager::class)->get($uid);
```
Comment (skjnldsv) [inline apps/files_sharing/lib/Cache.php:171]: DI?

File: lib/private/TemplateLayout.php [hunk 2]
```diff
@@ -123,7 +124,12 @@ public function __construct($renderAs, $appId = '') {
 					break;
 				}
 			}
-			$userDisplayName = \OC_User::getDisplayName();
+
+			$userDisplayName = false;
+			$user = \OC::$server->get(IUserSession::class)->getUser();
```
Comment (skjnldsv) [inline lib/private/TemplateLayout.php:129]: DI?

File: lib/private/TemplateLayout.php [hunk 3]
```diff
@@ -154,7 +160,11 @@ public function __construct($renderAs, $appId = '') {
 			\OC_Util::addStyle('guest');
 			$this->assign('bodyid', 'body-login');
 
-			$userDisplayName = \OC_User::getDisplayName();
+			$userDisplayName = false;
+			$user = \OC::$server->get(IUserSession::class)->getUser();
```
Comment (skjnldsv) [inline lib/private/TemplateLayout.php:164]: DI?

File: lib/private/TemplateLayout.php (hunk 2 above)
Comment (MorrisJobke) [reply lib/private/TemplateLayout.php:129]: Not used at all in this class 🙈

File: lib/private/TemplateLayout.php (hunk 3 above)
Comment (MorrisJobke) [reply lib/private/TemplateLayout.php:164]: Not used at all in this class 🙈

File: apps/files_sharing/lib/Cache.php (hunk 1 above)
Comment (MorrisJobke) [reply apps/files_sharing/lib/Cache.php]: Done ✅

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by MorrisJobke** [comment]
CI says no ... let me remove the DI part again as it causes a lot of trouble at other places :/

**Comment by ChristophWurst** [comment]
tech dept and not fully one -> let's move to 22

**Comment by rullzer** [comment]
Kill or continue?

**Comment by MorrisJobke** [comment]
> Kill or continue?

🔥 but the code and not the PR ... let me rebase and 🙏

File: apps/files_sharing/lib/SharedStorage.php [hunk 4]
```diff
@@ -384,7 +385,11 @@ public function getCache($path = '', $storage = null) {
 			return new FailedCache();
 		}
 
-		$this->cache = new \OCA\Files_Sharing\Cache($storage, $sourceRoot, $this->superShare);
+		$this->cache = new \OCA\Fwiles_Sharing\Cache(
```
Comment (splitt3r) [inline apps/files_sharing/lib/SharedStorage.php:388]:
```suggestion
		$this->cache = new \OCA\Files_Sharing\Cache(
```

**Review by splitt3r — CHANGES_REQUESTED** [review]

File: apps/files_sharing/lib/SharedStorage.php (hunk 4 above)
Comment (MorrisJobke) [reply apps/files_sharing/lib/SharedStorage.php:388]: 🙈 Thank you so much @splitt3r .... it's late already 🙈

**Comment by MorrisJobke** [comment]
Rebased.

**Review by splitt3r — APPROVED** [review]

**Comment by J0WI** [comment]
PHPUnit is not happy

**Comment by blizzz** [comment]
@MorrisJobke could you check the tests?

**Comment by MorrisJobke** [comment]
> @MorrisJobke could you check the tests?

Let's see if the latest change fixes it.

**Pull request merged**